scripts/zones/VeLugannon_Palace/mobs/Zipacna.lua

```diff
@@ -4,122 +4,523 @@
+        then
+            if
+                firstBlueDoor and
+                firstBlueDoor:getAnimation() == xi.anim.CLOSE_DOOR
+            then
+                mob:pathThrough(firstBlueToBasement, xi.path.flag.COORDS)
+            elseif
+                firstBlueDoor and
+                firstBlueDoor:getAnimation() == xi.anim.OPEN_DOOR
+            then
+                mob:pathThrough(firstBluetoBlueMiddle, xi.path.flag.COORDS)
+                mob:setLocalVar('[Zipacna]lastDoor', 1)
+            end
+        elseif -- coming from middle blue door to first blue door
+            pos.x >= -158.000 and
+            pos.y <= 1 and
+            pos.z >= 379 and
+            getLastDoor >= 1
+        then
+            if
+                firstBlueDoor and
+                firstBlueDoor:getAnimation() == xi.anim.CLOSE_DOOR
+            then
+                mob:pathThrough(firstBluetoBlueMiddle, xi.path.flag.COORDS)
+            elseif
+                firstBlueDoor and
+                firstBlueDoor:getAnimation() == xi.anim.OPEN_DOOR
+            then
+                mob:pathThrough(firstBlueToBasement, xi.path.flag.COORDS)
+            end
+        end
+    end
+
+    -- middle blue door
+    if
+    mob:checkDistance(middleBlueDoorCoord.x, middleBlueDoorCoord.y, middleBlueDoorCoord.z) <= 5 then
+        if
+            pos.x <= -81 and
+            pos.y <= 1 and
+            pos.z >= 459 and
+            getLastDoor == 1
+        then
+            if
+                middleBlueDoor and
+                middleBlueDoor:getAnimation() == xi.anim.CLOSE_DOOR
+            then
+                mob:pathThrough(middleBlueToFirstBlue, xi.path.flag.COORDS)
+            elseif
+                middleBlueDoor and
+                middleBlueDoor:getAnimation() == xi.anim.OPEN_DOOR
+            then
+                mob:pathThrough(middleBlueToMiddleYellow, xi.path.flag.COORDS)
+                mob:setLocalVar('[Zipacna]lastDoor', 2)
+            end
+        elseif
+            pos.x >= -76 and
+            pos.y <= 1 and
+            pos.z >= 460 and
+            getLastDoor >= 2
+        then
+            if
+                middleBlueDoor and
+                middleBlueDoor:getAnimation() == xi.anim.CLOSE_DOOR
+            then
+                mob:pathThrough(middleBlueToMiddleYellow, xi.path.flag.COORDS)
+            elseif
+                middleBlueDoor and
+                middleBlueDoor:getAnimation() == xi.anim.OPEN_DOOR
+            then
+                mob:pathThrough(middleBlueToFirstBlue, xi.path.flag.COORDS)
+            end
+        end
+    end
+    -- middle yellow door
+    if
+    mob:checkDistance(middleYellowDoorCoord.x, middleYellowDoorCoord.y, middleYellowDoorCoord.z) <= 5 then
+        if
+            pos.x <= 78 and
+            pos.y <= 1 and
+            pos.z <= 461 and
+            getLastDoor <= 3
+        then
+            print ("12check distance to middle yellow door is a success! from blue middle door")
```

Comment:
Remove prints

Reply:
Helping Hook out with this - So will comment on some of this on his behalf.
These have been removed.